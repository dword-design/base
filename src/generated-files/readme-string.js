import { endent, reduce, replace } from '@dword-design/functions'
import getProjectzReadmeSectionRegex from 'get-projectz-readme-section-regex'
import { readFileSync as safeReadFileSync } from 'safe-readfile'

import packageConfig from './package-config'
import replacements from './readme-replacements.config'

const readme =
  safeReadFileSync('README.md', 'utf8') ||
  endent`
  <!-- TITLE -->

  <!-- BADGES -->

  <!-- DESCRIPTION -->

  <!-- INSTALL -->

  <!-- LICENSE -->

`

export default replacements
  |> reduce((current, getReplacement, name) => {
    const sectionName = name.toUpperCase()
    return (
      current
      |> replace(
        getProjectzReadmeSectionRegex(sectionName),
        endent`
          <!-- ${sectionName}/ -->
          ${getReplacement(packageConfig)}
          <!-- /${sectionName} -->
        `
      )
    )
  }, readme)
