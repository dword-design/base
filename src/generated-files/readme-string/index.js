import { endent, reduce, replace } from '@dword-design/functions'
import getProjectzReadmeSectionRegex from 'get-projectz-readme-section-regex'
import { readFileSync as safeReadFileSync } from 'safe-readfile'

import packageConfig from '@/src/generated-files/package-config'

import replacements from './replacements'

const readme =
  safeReadFileSync('README.md', 'utf8') ||
  endent`
  <!-- TITLE -->

  <!-- BADGES -->

  <!-- DESCRIPTION -->

  <!-- INSTALL -->

  <!-- LICENSE -->

`

export default replacements
  |> reduce((current, getReplacement, name) => {
    const sectionName = name.toUpperCase()

    return (
      current
      |> replace(
        getProjectzReadmeSectionRegex(sectionName),
        endent`
          <!-- ${sectionName}/ -->
          ${getReplacement(packageConfig)}
          <!-- /${sectionName} -->
        `
      )
    )
  }, readme)
