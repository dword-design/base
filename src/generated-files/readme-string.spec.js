import { endent } from '@dword-design/functions'
import execa from 'execa'
import outputFiles from 'output-files'
import stealthyRequire from 'stealthy-require'
import withLocalTmpDir from 'with-local-tmp-dir'

export default {
  badges: () =>
    withLocalTmpDir(async () => {
      await execa.command('git init')
      await execa.command(
        'git remote add origin git@github.com:dword-design/bar.git'
      )
      await outputFiles({
        'README.md': endent`
        <!-- BADGES -->
        
      `,
        'package.json': JSON.stringify({ name: '@dword-design/foo' }),
      })
      const readmeString = stealthyRequire(require.cache, () =>
        require('./readme-string')
      )
      expect(readmeString).toEqual(endent`
        <!-- BADGES/ -->
        [![NPM version](https://img.shields.io/npm/v/@dword-design/foo.svg)](https://npmjs.org/package/@dword-design/foo)
        ![Linux macOS Windows compatible](https://img.shields.io/badge/os-linux%20%7C%C2%A0macos%20%7C%C2%A0windows-blue)
        [![Build status](https://img.shields.io/github/workflow/status/dword-design/bar/build)](https://github.com/dword-design/bar/actions)
        [![Coverage status](https://img.shields.io/coveralls/dword-design/bar)](https://coveralls.io/github/dword-design/bar)
        [![Dependency status](https://img.shields.io/david/dword-design/bar)](https://david-dm.org/dword-design/bar)
        ![Renovate enabled](https://img.shields.io/badge/renovate-enabled-brightgreen)

        <a href="https://www.buymeacoffee.com/dword">
          <img
            src="https://www.buymeacoffee.com/assets/img/guidelines/download-assets-sm-2.svg"
            alt="Buy Me a Coffee"
            height="32"
          >
        </a><a href="https://gitpod.io/#https://github.com/dword-design/bar">
          <img src="https://gitpod.io/button/open-in-gitpod.svg" alt="Open in Gitpod">
        </a>
        <a href="https://paypal.me/SebastianLandwehr">
          <img
            src="https://upload.wikimedia.org/wikipedia/commons/b/b5/PayPal.svg"
            alt="PayPal"
            height="30"
          >
        </a>
        <!-- /BADGES -->

      `)
    }),
  description: () =>
    withLocalTmpDir(async () => {
      await outputFiles({
        'README.md': endent`
        <!-- DESCRIPTION -->
        
      `,
        'package.json': JSON.stringify({ description: 'foo bar baz' }),
      })
      const readmeString = stealthyRequire(require.cache, () =>
        require('./readme-string')
      )
      expect(readmeString).toEqual(endent`
      <!-- DESCRIPTION/ -->
      foo bar baz
      <!-- /DESCRIPTION -->

    `)
    }),
  'existing content': () =>
    withLocalTmpDir(async () => {
      await outputFiles({
        'README.md': endent`
        <!-- DESCRIPTION -->

        This is a more detailed description

        <!-- LICENSE -->

      `,
        'package.json': JSON.stringify({
          author: 'dword-design',
          description: 'foo bar baz',
          license: 'MIT',
        }),
      })
      const readmeString = stealthyRequire(require.cache, () =>
        require('./readme-string')
      )
      expect(readmeString).toEqual(endent`
        <!-- DESCRIPTION/ -->
        foo bar baz
        <!-- /DESCRIPTION -->

        This is a more detailed description

        <!-- LICENSE/ -->
        ## License

        Unless stated otherwise all works are:

        Copyright &copy; Sebastian Landwehr <info@dword-design.de>

        and licensed under:

        [MIT License](https://opensource.org/licenses/MIT)
        <!-- /LICENSE -->

      `)
    }),
  install: () =>
    withLocalTmpDir(async () => {
      await outputFiles({
        'README.md': endent`
        <!-- INSTALL -->
        
      `,
        'package.json': JSON.stringify({ name: 'foo' }),
      })
      const readmeString = stealthyRequire(require.cache, () =>
        require('./readme-string')
      )
      expect(readmeString).toEqual(endent`
      <!-- INSTALL/ -->
      ## Install
      
      \`\`\`bash
      # NPM
      $ npm install foo

      # Yarn
      $ yarn add foo
      \`\`\`
      <!-- /INSTALL -->

    `)
    }),
  license: () =>
    withLocalTmpDir(async () => {
      await outputFiles({
        'README.md': endent`
        <!-- LICENSE -->
        
      `,
        'package.json': JSON.stringify({ license: 'MIT' }),
      })
      const readmeString = stealthyRequire(require.cache, () =>
        require('./readme-string')
      )
      expect(readmeString).toEqual(endent`
      <!-- LICENSE/ -->
      ## License

      Unless stated otherwise all works are:

      Copyright &copy; Sebastian Landwehr <info@dword-design.de>

      and licensed under:

      [MIT License](https://opensource.org/licenses/MIT)
      <!-- /LICENSE -->

    `)
    }),
  title: () =>
    withLocalTmpDir(async () => {
      await outputFiles({
        'README.md': endent`
        <!-- TITLE -->
        
      `,
        'package.json': JSON.stringify({
          name: 'foo',
        }),
      })
      const readmeString = stealthyRequire(require.cache, () =>
        require('./readme-string')
      )
      expect(readmeString).toEqual(endent`
      <!-- TITLE/ -->
      # foo
      <!-- /TITLE -->

    `)
    }),
}
