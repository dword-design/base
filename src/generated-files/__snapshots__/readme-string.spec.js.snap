// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readme-string badges 1`] = `
"<!-- BADGES/ -->
[![npm version](https://img.shields.io/npm/v/@dword-design/foo.svg)](https://npmjs.org/package/@dword-design/foo)
![Linux macOS Windows compatible](https://img.shields.io/badge/os-linux%20%7C%C2%A0macos%20%7C%C2%A0windows-blue)
[![Build status](https://github.com/dword-design/bar/workflows/build/badge.svg)](https://github.com/dword-design/bar/actions)
[![Coverage status](https://codecov.io/gh/dword-design/bar/branch/master/graph/badge.svg)](https://codecov.io/gh/dword-design/bar)
[![Dependency status](https://img.shields.io/david/dword-design/bar)](https://david-dm.org/dword-design/bar)
![Renovate enabled](https://img.shields.io/badge/renovate-enabled-brightgreen)

<a href=\\"https://gitpod.io/#https://github.com/dword-design/bar\\">
  <img src=\\"https://gitpod.io/button/open-in-gitpod.svg\\" alt=\\"Open in Gitpod\\">
</a><a href=\\"https://www.buymeacoffee.com/dword\\">
  <img
    src=\\"https://www.buymeacoffee.com/assets/img/guidelines/download-assets-sm-2.svg\\"
    alt=\\"Buy Me a Coffee\\"
    height=\\"32\\"
  >
</a><a href=\\"https://paypal.me/SebastianLandwehr\\">
  <img
    src=\\"https://dword-design.de/images/paypal.svg\\"
    alt=\\"PayPal\\"
    height=\\"32\\"
  >
</a><a href=\\"https://www.patreon.com/dworddesign\\">
  <img
    src=\\"https://dword-design.de/images/patreon.svg\\"
    alt=\\"Patreon\\"
    height=\\"32\\"
  >
</a>
<!-- /BADGES -->
"
`;
